src/init.cpp

```diff
@@ -1403,8 +1392,16 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     LogPrintf("* Using %.1f MiB for chain state database\n", cache_sizes.coins_db * (1.0 / 1024 / 1024));
     LogPrintf("* Using %.1f MiB for in-memory UTXO set (plus up to %.1f MiB of unused mempool space)\n", cache_sizes.coins * (1.0 / 1024 / 1024), nMempoolSizeMax * (1.0 / 1024 / 1024));
 
+    assert(!node.mempool);
+    assert(!node.chainman);
     bool fLoaded = false;
     while (!fLoaded && !ShutdownRequested()) {
+        node.mempool = std::make_unique<CTxMemPool>(node.fee_estimator.get(), check_ratio);
+        CTxMemPool& mempool = *Assert(node.mempool);
```

Comment:
in b7e44255af0fbd11f33b5425af9fba8809c2e2d9: Any reason to make mempool a reference? This conflicts with #22850.

It seems probably best to remove this line and leave the other line untouched. This will reduce the diff here and conflicts/code-churn elsewhere.

Reply:
Done 276a07f92925c9be82ecd2fe0504323ded1051a1